Libraries/Arduino_BNO055/BNO055_IMU.cpp

```diff
@@ -0,0 +1,138 @@
+            {
+                uint8_t raw_data[6];
+                i2c_read(I2C_ADDRESS, LIA_DATA_X_LSB, 6, raw_data);
+                data->accelAxes[0] = ((int16_t)raw_data[1] << 8) | raw_data[0];
+                data->accelAxes[1] = ((int16_t)raw_data[3] << 8) | raw_data[2];
+                data->accelAxes[2] = ((int16_t)raw_data[5] << 8) | raw_data[4];
+            }
+            break;
+        case GRAVITY:
+            {
+                uint8_t raw_data[6];
+                i2c_read(I2C_ADDRESS, GRV_DATA_X_LSB, 6, raw_data);
+                data->gravAxes[0] = ((int16_t)raw_data[1] << 8) | raw_data[0];
+                data->gravAxes[1] = ((int16_t)raw_data[3] << 8) | raw_data[2];
+                data->gravAxes[2] = ((int16_t)raw_data[5] << 8) | raw_data[4];
+            }
+            break;
+        case EUL:
+            {
+                uint8_t raw_data[6];
+                i2c_read(I2C_ADDRESS, EUL_HEADING_LSB, 6, raw_data);
+                data->eulerAngles[0] = ((int16_t)raw_data[1] << 8) | raw_data[0];
+                data->eulerAngles[1] = ((int16_t)raw_data[3] << 8) | raw_data[2];
+                data->eulerAngles[2] = ((int16_t)raw_data[5] << 8) | raw_data[4];
+            }
+            break;
+        case QUAT:
+            {
+                uint8_t raw_data[8];
+                i2c_read(I2C_ADDRESS, QUA_DATA_W_LSB, 8, raw_data);
+                data->quaternions[0] = ((int16_t)raw_data[1] << 8) | raw_data[0];
+                data->quaternions[1] = ((int16_t)raw_data[3] << 8) | raw_data[2];
+                data->quaternions[2] = ((int16_t)raw_data[5] << 8) | raw_data[4];
+                data->quaternions[3] = ((int16_t)raw_data[7] << 8) | raw_data[6];
+            }
+            break;
+    }
+}
+
+void update_data(Data *my_data)
+{
+    AxisData *data;
+    if (!(data = (AxisData *)malloc(sizeof(AxisData)))) {
+        return;
+    }
+
+    read_data(data, ACCEL);
+    for (size_t i = 0; i < 3; i++) {
+        my_data->accelAxes[i] = (float) data->accelAxes[i] / 100.;
+    }
+
+    read_data(data, GYRO);
+    for (size_t i = 0; i < 3; i++) {
+        my_data->gyroAxes[i] = (float) data->gyroAxes[i] / 16.;
+    }
+
+    read_data(data, GRAVITY);
+    for (size_t i = 0; i < 3; i++) {
+        my_data->gravAxes[i] = (float) data->gravAxes[i] / 100.;
+    }
+
+    read_data(data, EUL);
+    for (size_t i = 0; i < 3; i++) {
+        my_data->eulerAngles[i] = (float) data->eulerAngles[i] / 16.;
+    }
+
+    Quaternion q;
+    read_data(data, QUAT);
+    q.x = (float) data->quaternions[0] / 16384.;
+    q.y = (float) data->quaternions[1] / 16384.;
+    q.z = (float) data->quaternions[2] / 16384.;
+    q.w = (float) data->quaternions[3] / 16384.;
+
+    float yaw = atan2(2.0f*(q.y*q.z + q.w*q.x), q.x*q.x + q.y*q.y - q.z*q.z - q.w*q.w);
+    float pitch = -asin(2.0f*(q.y*q.w - q.x*q.z));
+    float roll = atan2(2.0f*(q.x*q.y + q.z*q.w), q.x*q.x - q.y*q.y - q.z*q.z + q.w*q.w);
+    
+    yaw *= -180.0f / M_PI;
```

Comment:
Is M_PI an arduino default constant or something? Otherwise not seeing it declared